src/core/handshaker/security/secure_endpoint.cc

```diff
@@ -724,6 +729,46 @@ class SecureEndpoint final : public EventEngine::Endpoint {
                SliceBuffer* data, WriteArgs args) {
       GRPC_LATENT_SEE_INNER_SCOPE("secure_endpoint write");
       tsi_result result;
+      if (grpc_core::IsSecureEndpointOffloadLargeWritesEnabled()) {
+        if (data->Length() == 0) return true;
+        grpc_core::MutexLock lock(&write_queue_mu_);
+        frame_protector_.TraceOp("WriteQ", data->c_slice_buffer());
+        if (!writing_.ok()) {
+          event_engine_->Run(
+              [on_write = std::move(on_writable),
+               status = writing_.status()]() mutable { on_write(status); });
+        }
+        if (*writing_) {
```

Comment:
Please add some comments for the different if conditions :)

Reply:
done